homeassistant/components/econet/sensor.py

```diff
@@ -1,113 +1,133 @@
+    ),
+    SensorEntityDescription(
+        key="todays_water_usage",
+        name="water_usage_today",
+        native_unit_of_measurement=UnitOfVolume.GALLONS,
+        device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="todays_energy_usage",
+        name="power_usage_today",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="alert_count",
+        name="alert_count",
+    ),
+    SensorEntityDescription(
+        key="wifi_signal",
+        name="wifi_signal",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="running_state",
+        name="running_state",
+    ),
+)
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up EcoNet sensor based on a config entry."""
 
-    equipment = hass.data[DOMAIN][EQUIPMENT][entry.entry_id]
-    sensors = []
-    all_equipment = equipment[EquipmentType.WATER_HEATER].copy()
-    all_equipment.extend(equipment[EquipmentType.THERMOSTAT].copy())
-
-    for _equip in all_equipment:
-        for name, attribute in SENSOR_NAMES_TO_ATTRIBUTES.items():
-            if getattr(_equip, attribute, None) is not None:
-                sensors.append(EcoNetSensor(_equip, name))
-        # This is None to start with and all device have it
-        sensors.append(EcoNetSensor(_equip, WIFI_SIGNAL))
+    data = hass.data[DOMAIN][EQUIPMENT][entry.entry_id]
+    equipment = data[EquipmentType.WATER_HEATER].copy()
+    equipment.extend(data[EquipmentType.THERMOSTAT].copy())
 
-    for water_heater in equipment[EquipmentType.WATER_HEATER]:
-        # These aren't part of the device and start off as None in pyeconet so always add them
-        sensors.append(EcoNetSensor(water_heater, WATER_USAGE_TODAY))
-        sensors.append(EcoNetSensor(water_heater, POWER_USAGE_TODAY))
+    sensors = [
+        EcoNetSensor(_equip, description)
+        for _equip in equipment
+        for description in SENSOR_TYPES
+        if getattr(_equip, description.key, False) is not False
+    ]
 
     async_add_entities(sensors)
 
 
 class EcoNetSensor(EcoNetEntity, SensorEntity):
     """Define a Econet sensor."""
 
-    def __init__(self, econet_device, device_name):
+    entity_description: SensorEntityDescription
+
+    def __init__(
+        self,
+        econet_device: Equipment,
+        description: SensorEntityDescription,
+    ) -> None:
         """Initialize."""
         super().__init__(econet_device)
+        self.entity_description = description
         self._econet = econet_device
```

Comment:
This is set inside `EcoNetEntity`, so no need to set it again here.
```suggestion
```

Reply:
Makes sense.